### ISSUE TITLE
Address becomes different with different chainId

### ISSUE THREAD

**Issue by V-Staykov** [chainapsis/keplr-wallet#239]
**Describe the bug**
After suggesting several chains to the walled, that had the same config with one difference - the chainId, I found that the address of the wallet is different for one of the chainIds, which is very strange. Also this happens for more than one account, but not on all. I couldn't find any reason for certain accounts to have different address on that one chainId and others not.

**To Reproduce**
1. Add a wallet with the following mnemonic to your keplr wallet from the "Import existing wallet" option:
`budget purchase bomb comfort olive crisp canyon accident noble bounce endless dignity expect pottery discover wrestle rack major sphere light village cloth around phone`
2. Suggest 2 chains - one with chainId **cudos-network** and the other with w/e chainId:
```
var config = {
      rpc: 'https://abc.org',
      rest: 'https://abc.org',
      chainName: 'Cudos Network',
      chainId: 'cudos-network',
      currencies: [{
        coinDenom: 'CUDOS',
        coinMinimalDenom: 'acudos',
        coinDecimals: 18,
        coinGeckoId: 'cudos'
      }],
      stakeCurrency: {
        coinDenom: 'CUDOS',
        coinMinimalDenom: 'acudos',
        coinDecimals: 18,
        coinGeckoId: 'cudos'
      },
      feeCurrencies: [{
        coinDenom: 'CUDOS',
        coinMinimalDenom: 'acudos',
        coinDecimals: 18,
        coinGeckoId: 'cudos'
      }],
      bip44: { coinType: 118 },
      bech32Config: {
        bech32PrefixAccAddr: 'cudos',
        bech32PrefixAccPub: 'cudospub',
        bech32PrefixValAddr: 'cudosvaloper',
        bech32PrefixValPub: 'cudosvaloperpub',
        bech32PrefixConsAddr: 'cudosvalcons',
        bech32PrefixConsPub: 'cudosvalconspub'
      }
    }
window.keplr.experimentalSuggestChain(config)

var config = {
      rpc: 'https://abc.org',
      rest: 'https://abc.org',
      chainName: 'Cudos Network 2',
      chainId: 'something-different-id-of-the',
      currencies: [{
        coinDenom: 'CUDOS',
        coinMinimalDenom: 'acudos',
        coinDecimals: 18,
        coinGeckoId: 'cudos'
      }],
      stakeCurrency: {
        coinDenom: 'CUDOS',
        coinMinimalDenom: 'acudos',
        coinDecimals: 18,
        coinGeckoId: 'cudos'
      },
      feeCurrencies: [{
        coinDenom: 'CUDOS',
        coinMinimalDenom: 'acudos',
        coinDecimals: 18,
        coinGeckoId: 'cudos'
      }],
      bip44: { coinType: 118 },
      bech32Config: {
        bech32PrefixAccAddr: 'cudos',
        bech32PrefixAccPub: 'cudospub',
        bech32PrefixValAddr: 'cudosvaloper',
        bech32PrefixValPub: 'cudosvaloperpub',
        bech32PrefixConsAddr: 'cudosvalcons',
        bech32PrefixConsPub: 'cudosvalconspub'
      }
    }
window.keplr.experimentalSuggestChain(config)
```

3. Open Keplr and switch between the two new chains - they get different addresses, namely:
**cudos19kas7ua8z5n2m0j6zul8zknhtttket3np47upl** for chainId cudos-network and 
**cudos1h9fl2mq0anv33h3sxprkx5k655x3uruydj3cp6** for any other chainId

**Expected behavior**
The two addresses should be the same

**Screenshots**

**Device details (please complete the following information):**
 - OS: Windows
 - Browser: Version 1.33.106 Chromium: 96.0.4664.110 (Official Build) (64-bit).0.4664.110 (Official Build) (64-bit)
 - Keplr Version: 0.9.10
 - Hardware Wallet: n/a

**Additional context**

**Comment by Thunnini** [comment]
If you create an account again with the same mnemonic, do the addresses come out differently?

Internally, keplr remembers the coin type used in the account per chain id. If you register a chain with a different coin type, and delete it, and then add a chain with another coin type and with the same chain id, the account remembers the previous coin type and uses that coin type, so the address is may come out differently.

**Comment by V-Staykov** [comment]
> If you create an account again with the same mnemonic, do the addresses come out differently?

No, but I did try adding new chains with different coin types and I managed to reproduce the problem with different chainIds. So what happened I guess is that the chain was first suggested with a cointype 60 and then even after deleting it and suggesting it again with cointype 118, the different addresses stayed. 

Is this an expected behaviour?

**Comment by Thunnini** [comment]
In fact, it is not expected behavior. Because some chains can give users a choice among several coin types, it is a result of the logic for that. Do you think the above case should be corrected? @dogemos @delivan
